examples/notebooks/beam-ml/rag_usecase/opensearch_enrichment.py

```diff
@@ -0,0 +1,138 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module implements enrichment classes to implement semantic search on opensearch Vector DB.
+
+
+opensearch :Enrichment Handler
+-----------------
+:class:`opensearchEnrichmentHandler` is a ``EnrichmentSourceHandler`` that performs enrichment/search
+by fetching the similar text to the user query/prompt from the knowledge base (opensearch vector DB) and returns
+the similar text along with its embeddings as Beam.Row Object. 
+
+Example usage::
+  opensearch_handler = opensearchEnrichmentHandler(opensearch_host='127.0.0.1', opensearch_port=6379)
+  
+  pipeline | Enrichment(opensearch_handler)
+
+No backward compatibility guarantees. Everything in this module is experimental.
+"""
+
+
+import logging
+
+
+import numpy as np
+from opensearchpy import OpenSearch
+
+
+import apache_beam as beam
+from apache_beam.transforms.enrichment import EnrichmentSourceHandler
+from apache_beam.transforms.enrichment_handlers.utils import ExceptionLevel
+
+__all__ = [
+    'OpenSearchEnrichmentHandler',
+]
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OpenSearchEnrichmentHandler(EnrichmentSourceHandler[beam.Row, beam.Row]):
+  """A handler for :class:`apache_beam.transforms.enrichment.Enrichment`
+  transform to interact with opensearch vector DB.
+
+  Args:
+    opensearch_host (str): opensearch Host to connect to opensearch DB
+    opensearch_port (int): opensearch Port to connect to opensearch DB
+    index_name (str): Index Name created for searching in opensearch DB
+    vector_field (str): vector field to compute similarity score in vector DB
+    return_fields (list): returns list of similar text and its embeddings
+    hybrid_fields (str): fields to be selected
+    k (int): Value of K in KNN algorithm for searching in opensearch
+  """
```

Comment:
These should be documented as args to `__init__` not the class declaration

Reply:
done